File: tui/model/entry.go
package model

type Entry struct {
	ID                 string   `json:"id"`
	RawText            string   `json:"raw_text"`
	DurationMinutes    *int     `json:"duration_minutes"`
	WorkTypeID         *string  `json:"work_type_id"`
	TimeCodeID         *string  `json:"time_code_id"`
	Description        *string  `json:"description"`
	EntryDate          string   `json:"entry_date"`
	ConfidenceDuration *float64 `json:"confidence_duration"`
	ConfidenceWorkType *float64 `json:"confidence_work_type"`
	ConfidenceTimeCode *float64 `json:"confidence_time_code"`
	ConfidenceOverall  *float64 `json:"confidence_overall"`
	Status             string   `json:"status"`
	ParseError         *string  `json:"parse_error"`
	ManuallyCorrect    bool     `json:"manually_corrected"`
	Locked             bool     `json:"locked"`
	CorrectedAt        *Time    `json:"corrected_at"`
	CreatedAt          Time     `json:"created_at"`
	ParsedAt           *Time    `json:"parsed_at"`
	UpdatedAt          Time     `json:"updated_at"`
}

File: backend/src/tact/schemas/entry.py
from datetime import date, datetime

from pydantic import BaseModel


class EntryCreate(BaseModel):
    raw_text: str
    entry_date: date | None = None


class EntryUpdate(BaseModel):
    raw_text: str | None = None
    duration_minutes: int | None = None
    work_type_id: str | None = None
    time_code_id: str | None = None
    description: str | None = None
    entry_date: date | None = None
    status: str | None = None


class EntryResponse(BaseModel):
    id: str
    raw_text: str
    duration_minutes: int | None
    work_type_id: str | None
    time_code_id: str | None
    description: str | None
    entry_date: date | None
    confidence_duration: float | None
    confidence_work_type: float | None
    confidence_time_code: float | None
    confidence_overall: float | None
    status: str
    parse_error: str | None
    parse_notes: str | None
    manually_corrected: bool
    locked: bool
    corrected_at: datetime | None
    created_at: datetime
    parsed_at: datetime | None
    updated_at: datetime

    model_config = {"from_attributes": True}
